src/app/data_harvester/memory/general/heim.rs

```diff
@@ -7,13 +7,17 @@ pub async fn get_mem_data(
 ) -> (
     crate::utils::error::Result<Option<MemHarvest>>,
     crate::utils::error::Result<Option<MemHarvest>>,
+    crate::utils::error::Result<Option<MemHarvest>>,
 ) {
     use futures::join;
 
     if !actually_get {
-        (Ok(None), Ok(None))
+        (Ok(None), Ok(None), Ok(None))
+    } else if cfg!(feature = "zfs") && (cfg!(target_os = "linux") || cfg!(target_os = "freebsd")) {
+        join!(get_ram_data(), get_swap_data(), get_arc_data())
     } else {
-        join!(get_ram_data(), get_swap_data())
+        let futures = join!(get_ram_data(), get_swap_data());
+        (futures.0, futures.1, Ok(None)) // stub get_arc_data
```

Comment:
Again, bit of a nit but I would prefer to explicitly do this check on compilation rather than doing a runtime check when it isn't really needed.

Another nit: `cfg!(feature = "zfs") && (cfg!(target_os = "linux") || cfg!(target_os = "freebsd")) ` can be a bit confusing at a glance, please use parentheses to indicate grouping/precedence if you keep this.

Reply:
Removed this runtime check.

I could remove the function call(s) all together (if desired) with compile time checks for target platforms instead of returning zero total ram and zero used arc for invalid platforms. (It may help performance?)

My main concern is that sprinkling in more platform specific code may make it more difficult for support of the other targets later ( more places in the code to change for macOS/Windows ). afaik there is a port of zfs for windows but I am unsure of its status. 

Thoughts?